addon/components/frost-link.js

```diff
@@ -243,11 +243,46 @@ export default LinkComponent.extend(SpreadMixin, PropTypeMixin, {
 
   // == Lifecycle Hooks =======================================================
 
-  /* Ember.Component method */
+  /**
+   * Explicitly overrides the didReceiveAttrs in link-to
+   * https://github.com/emberjs/ember.js/blob/v2.10.0/packages/ember-glimmer/lib/components/link-to.js#L790
+   *
+   * This allows named parameters to be passed to the link instead of positional
+   * params.  However, because the link-to expects positional params the named
+   * parameters must be set back against the component as a positional param
+   * array in the correct order (text, route, queryParams, models).  Once set
+   * we're free to hand control back to the parent function and it will react
+   * as if we used the original link-to interface.
+   */
+  didReceiveAttrs () {
```

Comment:
If we're discussing in ember 2.8 context, the htmlbar link-to component was still processing the positional params in **willRender** hook. It has been moved to **didReceiveAttrs** since 2.10 when ember switch from htmlbar to glimmar.